lib/mastermind.ex

```diff
@@ -82,4 +89,18 @@ defmodule Mastermind do
     1..@width
     |> Enum.map(fn _ -> Enum.random(possibles) end)
   end
+
+  def score_guess(guess, target) do
+    dodges = length(target) - length(target -- guess)
+    hits = matches(guess, target)
+    %{:reds => hits, :whites => dodges - hits}
+  end
+
+  def matches([], []), do: 0
```

Comment:
A slightly tidier variant of this is

~~~ elixir
def matches([], []), do: 0
def matches([h | t], [ h | t ]), do: 1 + matches(t, t)
def matches([_ | t], [ _ | t ]), do:   matches(t, t)
~~~

And, only because this code is better than 90% of the undergrads I teach, I'm going to be a little picky about the function name. When I first read `matches` I assumed it was returning some representation of the matched pegs. I wonder if `count_matches` or `number_of_matches` would be clearer?

Like I said: that's picky.